hive.go

```diff
@@ -836,13 +836,16 @@ func (c *Cursor) FetchOne(ctx context.Context, dests ...interface{}) {
 			if !ok {
 				d, ok := dests[i].(**int8)
 				if !ok {
-					c.Err = errors.Errorf("Unexpected data type %T for value %v (should be %T) index is %v", dests[i], c.queue[i].ByteVal.Values[c.columnIndex], c.queue[i].ByteVal.Values[c.columnIndex],i)
+					c.Err = errors.Errorf("Unexpected data type %T for value %v (should be %T) index is %v", dests[i], c.queue[i].ByteVal.Values[c.columnIndex], c.queue[i].ByteVal.Values[c.columnIndex], i)
 					return
 				}
 
 				if isNull(c.queue[i].ByteVal.Nulls, c.columnIndex) {
 					*d = nil
 				} else {
+					if *d == nil {
+						*d = new(int8)
+					}
 					**d = c.queue[i].ByteVal.Values[c.columnIndex]
```

Comment:
If you don't want to init every pointer in the struct, then let's just not assign anything to the nil pointer, you can do:

```
                                           if *d != nil {
					        **d = c.queue[i].ByteVal.Values[c.columnIndex]
					}
```

Reply:
II mean the assignment should succeed in this sutuation. The standard json/sql package init the pointer too. For me, init the pointer is extra work. What's your opinion?
```go
package main

import (
	"encoding/json"
	"fmt"
)

func main() {
	var p *map[string]interface{}
	j := `{"1":"2"}`
	json.Unmarshal([]byte(j), &p)
	fmt.Println(*p)
}

```

```go
package main

import (
	"database/sql"
	"fmt"

	_ "github.com/go-sql-driver/mysql"
)

func main() {
	db, err := sql.Open("mysql", "root:@tcp(localhost:3306)/train?timeout=5s")
	if err != nil {
		panic(err)
	}
	query := "select 1 from user limit 1"
	var ptr *int
	if err := db.QueryRow(query).Scan(&ptr); err != nil && err != sql.ErrNoRows {
		panic(err)
	}
	fmt.Println(*ptr)
}

Reply:
hmm, I guess you're right, I was worried about gohive unexpectedly allocating memory, but if the user is already requesting those fields it must be because they are going to be read.